InternetHospital.WebApi/Controllers/DoctorsController.cs

```diff
@@ -50,5 +54,60 @@ public IEnumerable<SpecializationModel> GetSpecializations()
         {
             return _doctorService.GetAvailableSpecialization();
         }
+
+        [HttpPut("update")]
+        [Authorize]
+        public async Task<IActionResult> UpdateDoctor([FromForm(Name = "PassportURL")]IFormFileCollection passport,
+            [FromForm(Name = "DiplomaURL")]IFormFileCollection diploma,
+            [FromForm(Name = "LicenseURL")]IFormFileCollection license)
+        {
+
+            var doctorModel = new DoctorProfileModel
+            {
+                FirstName = Request.Form["FirstName"],
+                SecondName = Request.Form["SecondName"], // First letter to Uper other tolower case !!!
+                ThirdName = Request.Form["ThirdName"],
+                PhoneNumber = Request.Form["PhoneNumber"],
+                BirthDate = DateTime.Parse(Request.Form["BirthDate"]),
+                Address = Request.Form["Address"],
+                Specialization = Request.Form["Specialization"],
+            };
+
+            TryValidateModel(doctorModel);
+
+            if (ModelState.IsValid)
+            {
+                if (int.TryParse(User.Identity.Name, out int userId))
+                {
+                    var result = await _doctorService.UpdateDoctorInfo(doctorModel, userId, passport, diploma, license);
+                    return result ? (IActionResult)Ok() : BadRequest(new { message = "Error during updating!" });
+                }
+                else
+                {
+                    return BadRequest(new { message = "Error with user ID!" });
+                }
+            }
+            return BadRequest();
+        }
+
+        [HttpGet("getProfile")]
+        [Authorize]
```

Comment:
Role-based authorize is necessary here. Here will be NullReferenceException if patient call this method.